src/code/z_schedule.c

```diff
@@ -0,0 +1,311 @@
+#include "global.h"
+
+s32 Schedule_FlagCheckS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdFlagCheckS* cmd = (ScheduleCmdFlagCheckS*)*script;
+    u16 flag = (cmd->flagByte << 8) | cmd->flagMask;
+
+    if (gSaveContext.weekEventReg[flag >> 8] & (flag & 0xFF)) {
+        *script = *script + cmd->offset;
+    }
+
+    return false;
+}
+
+s32 Schedule_FlagCheckL(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdFlagCheckL* cmd = (ScheduleCmdFlagCheckL*)*script;
+    u16 flag = (cmd->flagByte << 8) | cmd->flagMask;
+
+    if (gSaveContext.weekEventReg[flag >> 8] & (flag & 0xFF)) {
+        *script = *script + (s16)((cmd->offsetH << 8) | cmd->offsetL);
+        ;
+    }
+
+    return false;
+}
+
+s32 Schedule_TimeRangeCheckS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    s32 inRange = 0;
+    ScheduleCmdTimeRangeCheckS* cmd = (ScheduleCmdTimeRangeCheckS*)*script;
+    f32 f;
+    u16 start;
+    u16 end;
+    u16 now;
+
+    f = cmd->startHr * 60.0f;
```

Comment:
This is close to how `CLOCK_TIME` works, but this has to use a float temp as well on be multiple lines. I considered a macro for this code pattern but wasn't sure it would help readability.

Reply:
A macro would be nice. Do you know if this pattern is used in more places?

Reply:
Not that I know of outside this file. I mention it because it is very similar to [`CLOCK_TIME`](https://github.com/zeldaret/mm/blob/850823c2ee89aad0f0e53aa70777af11af6d5433/include/macros.h#L53). However, thinking on it more, It doesn't make sense as a macro. Mainly because of that float temp. It doesn't work if you move the temp into it's own `{}` block; it needs to be reused. That means the macro would be something like
```
#define SCHEDULE_CALC_TIME(hour, minute, dest, temp) \
    (temp) = (hour) * 60.0f; \
    (temp) += (minute); \
    (dest) = (temp) * 45.5f;
```
and honestly that's a bit weird to me, but not the worst.

As an aside, that `45.5f` literal stands out and is probably something close to `0x10000 / (24 * 60)`. I'll play with it.

Reply:
Just as a note #514 is adding some float versions to the CLOCK_TIME macros. They probably won't work because of the temp, but might give some ideas.

Reply:
Figured out that `45.5f = (0x10000 / 60 / 24.0f)`. However, I couldn't get it into a form that didn't mess with regalloc. Left in in a comment instead and made the macro.

Reply:
Never mind, just tried it again and it worked. I was testing it on a function that used it twice and I think it was messing it up.

The ones in #514 don't match exactly, because in this case the truncation happens at specific point. The `0x10000 / 60` needs to be truncated before the conversion to a float.

Reply:
Oh right, this file uses several "wrong way round" time conversions from most of the files I'm aware of, at least.